persistence/src/commonMain/kotlin/com/wire/kalium/persistence/dao/message/MessageMapper.kt

```diff
@@ -46,92 +47,130 @@ object MessageMapper {
         memberChangeType: MessageEntity.MemberChangeType?,
         isMentioningSelfUser: Boolean,
         isQuotingSelfUser: Boolean?,
+        isEphemeral: Boolean,
+        isGroupConversation: Boolean,
         text: String?,
         assetMimeType: String?,
         selfUserId: QualifiedIDEntity?,
         senderUserId: QualifiedIDEntity?
-    ) = when (contentType) {
-        MessageEntity.ContentType.TEXT -> when {
-            isSelfMessage -> MessagePreviewEntityContent.Text(
-                senderName = senderName,
-                messageBody = text.requireField("text")
-            )
-
-            (isQuotingSelfUser ?: false) -> MessagePreviewEntityContent.QuotedSelf(
-                senderName = senderName,
-                messageBody = text.requireField("text")
+    ): MessagePreviewEntityContent {
+        return if (isEphemeral) {
+            MessagePreviewEntityContent.Ephemeral(isGroupConversation)
+        } else {
+            mapContentType(
+                contentType,
+                senderName,
+                isSelfMessage,
+                memberChangeList,
+                memberChangeType,
+                isMentioningSelfUser,
+                isQuotingSelfUser,
+                text,
+                assetMimeType,
+                selfUserId,
+                senderUserId
             )
+        }
+    }
```

Comment:
Can we cover this mapping with unit test ?

Reply:
I pushed like seconds before you reviewed the unit tests I should have asked for review when it was really ready before